src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/json/JsonResponseData.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright <2020> Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.integ_test.json;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.integ_test.json.JsonResponseField.Type.Constants;
+import com.google.gson.annotations.SerializedName;
+
+public class JsonResponseData {
+  private static final String FIELDS = "fields";
+  private static final String RECORDS = "records";
+  @SerializedName(FIELDS)
+  private JsonResponseField[] fields;
+  @SerializedName(RECORDS)
+  private String[][] records;
+
+  public JsonResponseData(JsonResponseField[] fields, String[][] records) {
+    this.fields = fields;
+    this.records = records;
+  }
+
+  public int getFieldDimensionSize() {
+    return fields.length;
+  }
+
+  public int getRecordSize() {
+    return records.length;
+  }
+
+  public JsonResponseField getField(int index) throws IndexOutOfBoundsException {
+    return fields[index];
+  }
+
+  public String getRecord(int index, String fieldName) throws Exception {
+    for (int i = 0; i < getFieldDimensionSize(); i++) {
+      if (fieldName.equals(fields[i].getName())) {
+        return records[index][i];
+      }
+    }
+    throw new IllegalArgumentException();
```

Comment:
in which scenario this `IllegalArgumentException` will be thrown?

Reply:
this exception will be thrown if the record retrieved from PA does not have the "field" we look for